### ISSUE TITLE
iotile-gateway unhandled error when closing

### ISSUE THREAD

**Issue by mattrunchey** [iotile/coretools#795]
`iotile-gateway` running on the latest `master` branch code (asyncio), and connecting to it with websockets works, but closing seems to be happening a bit uncleanly from the iotile tool:

```
(v.8) Matts-Macbook-Pro:~ mrunchey$ iotile hw --port=ws:localhost:5120/iotile/v1
(HardwareManager) quit
Exception ignored in: <generator object WebSocketCommonProtocol.close_connection at 0x107e2f0a0>
Traceback (most recent call last):
  File "/Users/mrunchey/v.8/lib/python3.5/site-packages/websockets/protocol.py", line 1054, in close_connection
    self.writer.close()
  File "/usr/local/Cellar/python35/3.5.6_2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/asyncio/streams.py", line 306, in close
    return self._transport.close()
  File "/usr/local/Cellar/python35/3.5.6_2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/asyncio/selector_events.py", line 622, in close
    self._loop.call_soon(self._call_connection_lost, None)
  File "/usr/local/Cellar/python35/3.5.6_2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/asyncio/base_events.py", line 573, in call_soon
    self._check_closed()
  File "/usr/local/Cellar/python35/3.5.6_2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/asyncio/base_events.py", line 357, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
```

**Comment by mattrunchey** [comment]
Error came from a task not being cancelled due to an exception that happened before the task:

```
self._connection_task = self._loop.add_task(self._manage_connection(self._con), name=name)
```

`self._con` isn't a `Task` object, so `manage_connection` errored when it tried to cancel that `task` object (`  task.cancel_subtasks()`). 

However, this wasn't clear because the cancelled tasks aren't designed to raise their exceptions or print them, but we can add this logic to the event_loop for easier debugging when a task fails due to an exception that isn't a `CancelledError`. Essentially, we can 

```
            outcomes = await asyncio.wait_for(finished, timeout=self._stop_timeout)
```

And then iterate over the outcomes to see if any are of an Exception type that isn't CancelledError.

Working on a PR for this now.

**Comment by mattrunchey** [comment]
Fixed in #796
